File: library/src/main/java/com/flipkart/okhttpstats/handler/OnStatusCodeAwareResponseListener.java
package com.flipkart.okhttpstats.handler;

import android.net.NetworkInfo;

import com.flipkart.okhttpstats.model.RequestStats;

/**
 *
 */
public interface OnStatusCodeAwareResponseListener {
    void onResponseServerSuccess(NetworkInfo networkInfo, RequestStats requestStats);

    void onResponseServerError(NetworkInfo networkInfo, RequestStats requestStats);

    void onResponseNetworkError(NetworkInfo info, RequestStats requestStats, Exception e);
}

File: library/src/main/java/com/flipkart/okhttpstats/handler/OnStatusCodeAwareResponseListener.kt
package com.flipkart.okhttpstats.handler

import com.flipkart.okhttpstats.model.RequestStats

interface OnStatusCodeAwareResponseListener {
    fun onResponseServerSuccess(networkInfo: android.net.NetworkInfo?, requestStats: RequestStats)

    fun onResponseServerError(networkInfo: android.net.NetworkInfo?, requestStats: RequestStats)

    fun onResponseNetworkError(info: android.net.NetworkInfo?, requestStats: RequestStats, e: Exception)
}